singletons-th/src/Data/Singletons/TH/Promote.hs

```diff
@@ -957,6 +966,56 @@ information whatsoever. Until the aformentioned Haddock issue is resolved, we
 work around this limitation by generating the redundant argument and kind
 information anyway. Thankfully, this is simple to accomplish, as we already
 compute this information to begin with.
+
+Note [Generating type families with the correct arity]
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+As of GHC 9.10, GHC no longer performs arity inference when kind-checking type
+family declarations with standalone kind signatures. This is an important
+consideration when promoting functions with top-level type signatures. For
+example, we would not want to take this definition:
+
+  f :: Maybe a
+  f = Nothing
```

Comment:
Did you mean to write 
```
f :: Either a Bool
f = Right True
```
here, like in the PR description?

Reply:
Oops! I wrote this up, changed my mind about what example to use, and then forget to modify this part. Thanks for noticing this—I've changed it to be consistent with the rest of the documentation.